program/src/state/rules.rs

```diff
@@ -320,16 +320,20 @@ impl Rule {
             Rule::PubkeyListMatch { pubkeys, field } => {
                 msg!("Validating PubkeyListMatch");
 
-                let key = match payload.get_pubkey(field) {
-                    Some(pubkey) => pubkey,
-                    _ => return (false, RuleSetError::MissingPayloadValue.into()),
-                };
+                let fields = field.split('|').collect::<Vec<&str>>();
 
-                if pubkeys.iter().any(|pubkey| pubkey == key) {
-                    (true, self.to_error())
-                } else {
-                    (false, self.to_error())
+                for field in fields {
+                    let key = match payload.get_pubkey(&field.to_owned()) {
+                        Some(pubkey) => pubkey,
+                        _ => return (false, RuleSetError::MissingPayloadValue.into()),
+                    };
+
+                    if pubkeys.iter().any(|pubkey| pubkey == key) {
+                        return (true, self.to_error());
+                    }
```

Comment:
Shouldn't this have an else? If we found one `field` but didn't pass, we should not go onto the next `field` in the field loop, correct?

Reply:
We're replacing the multiple fields nested in an ANY rule with this so it's an implicit OR operation. We can add an AND later but I wanted to get this out quick for integration work.

Reply:
Ok I think I'm too tired to get the logic.  Sounds like you guys get it, thanks!

Reply:
Yeah now that I got some sleep its easy to see how it works.  Yeah I agree with @austbot we should also add an `&` operator for this method, and we should also consider adding it to the remaining rules for completeness.